commit ec1ad281062fdd632b3af58d9242fde30cdb4b29
Author: kirmaniarslan <arslankirmani9600@gmail.com>
Date:   Mon Nov 17 11:19:03 2014 +0100

    Add support for msm7x27 and msm7x27a board

diff --git a/core/qcom_target.mk b/core/qcom_target.mk
index 476548f..71a155c 100644
--- a/core/qcom_target.mk
+++ b/core/qcom_target.mk
@@ -18,7 +18,7 @@ ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
     TARGET_ENABLE_QC_AV_ENHANCEMENTS := true
 
     # Enable DirectTrack for legacy targets
-    ifneq ($(filter msm7x30 msm8660 msm8960,$(TARGET_BOARD_PLATFORM)),)
+    ifneq ($(filter msm7x27 msm7x27a msm7x30 msm8660 msm8960,$(TARGET_BOARD_PLATFORM)),)
     ifeq ($(BOARD_USES_LEGACY_ALSA_AUDIO),true)
         TARGET_GLOBAL_CFLAGS += -DQCOM_DIRECTTRACK
         TARGET_GLOBAL_CPPFLAGS += -DQCOM_DIRECTTRACK
diff --git a/core/qcom_utils.mk b/core/qcom_utils.mk
index 1177ad4..7dbf13f 100755
--- a/core/qcom_utils.mk
+++ b/core/qcom_utils.mk
@@ -1,5 +1,7 @@
 # Board platforms lists to be used for
 # TARGET_BOARD_PLATFORM specific featurization
+QCOM_BOARD_PLATFORMS += msm7x27
+QCOM_BOARD_PLATFORMS += msm7x27a
 QCOM_BOARD_PLATFORMS += msm7x30
 QCOM_BOARD_PLATFORMS += msm8226
 QCOM_BOARD_PLATFORMS += msm8610
