diff --git a/core/java/android/webkit/CookieManagerClassic.java b/core/java/android/webkit/CookieManagerClassic.java
index 36159e1..232d8d2 100644
--- a/core/java/android/webkit/CookieManagerClassic.java
+++ b/core/java/android/webkit/CookieManagerClassic.java
@@ -135,6 +135,11 @@ protected Void doInBackground(Void... none) {
     }
 
     @Override
+    public int countCookies() {
+        return nativeCountCookies();
+    }
+
+    @Override
     public void removeAllCookie() {
         nativeRemoveAllCookie();
     }
@@ -181,4 +186,5 @@ protected void setAcceptFileSchemeCookiesImpl(boolean accept) {
     private static native void nativeFlushCookieStore();
     private static native boolean nativeAcceptFileSchemeCookies();
     private static native void nativeSetAcceptFileSchemeCookies(boolean accept);
+    private static native int nativeCountCookies();
 }
-- 
1.9.3
