diff --git a/libgralloc/framebuffer.cpp b/libgralloc/framebuffer.cpp
index 2005540..efa998e 100644
--- a/libgralloc/framebuffer.cpp
+++ b/libgralloc/framebuffer.cpp
@@ -152,8 +152,9 @@ static int fb_post(struct framebuffer_device_t* dev, buffer_handle_t buffer)
 static int fb_compositionComplete(struct framebuffer_device_t* dev)
 {
     // TODO: Properly implement composition complete callback
+#ifdef ANCIENT_GL
     glFinish();
-
+#endif
     return 0;
 }
 
@@ -281,7 +282,7 @@ int mapFrameBufferLocked(struct private_module_t* module)
     uint32_t line_length = (info.xres * info.bits_per_pixel / 8);
     info.yres_virtual = (size * numberOfBuffers) / line_length;
 
-#ifndef NO_HW_VSYNC
+#ifdef MSMFB_METADATA_SET
     struct msmfb_metadata metadata;
 
     metadata.op = metadata_op_base_blend;
@@ -314,8 +315,13 @@ int mapFrameBufferLocked(struct private_module_t* module)
 
     float xdpi = (info.xres * 25.4f) / info.width;
     float ydpi = (info.yres * 25.4f) / info.height;
-    //The reserved[3] field is used to store FPS by the driver.
-    float fps  = info.reserved[3] & 0xFF;
+#ifndef REFRESH_RATE
+    //The reserved[4] field is used to store FPS by the driver.
+    float fps  = info.reserved[4];
+#else
+    float fps = REFRESH_RATE;
+#warning "refresh rate set via makefile to REFRESH_RATE"
+#endif
 
     ALOGI("using (fd=%d)\n"
           "id           = %s\n"
diff --git a/libhwcomposer/Android.mk b/libhwcomposer/Android.mk
index 2372847..c018e7e 100644
--- a/libhwcomposer/Android.mk
+++ b/libhwcomposer/Android.mk
@@ -7,7 +7,7 @@ LOCAL_MODULE_TAGS             := optional
 LOCAL_C_INCLUDES              := $(common_includes) $(kernel_includes)
 LOCAL_SHARED_LIBRARIES        := $(common_libs) libEGL liboverlay libgenlock \
                                  libhwcexternal libqdutils libhardware_legacy \
-                                 libdl libmemalloc libhwcservice
+                                 libdl libmemalloc libhwcservice libGLESv1_CM
 
 LOCAL_CFLAGS                  := $(common_flags) -DLOG_TAG=\"hwcomposer\"
 LOCAL_ADDITIONAL_DEPENDENCIES := $(common_deps)
diff --git a/libhwcomposer/hwc_copybit.cpp b/libhwcomposer/hwc_copybit.cpp
index af69e63..35f9cb1 100644
--- a/libhwcomposer/hwc_copybit.cpp
+++ b/libhwcomposer/hwc_copybit.cpp
@@ -21,6 +21,7 @@
 #define DEBUG_COPYBIT 0
 #include <copybit.h>
 #include <genlock.h>
+#include <GLES/gl.h>
 #include "hwc_copybit.h"
 #include "comptype.h"
 #include "egl_handles.h"
@@ -189,6 +190,14 @@ bool CopyBit::draw(hwc_context_t *ctx, hwc_display_contents_1_t *list, EGLDispla
         return -1;
     }
 
+#ifndef ANCIENT_GL
+    // Invoke a glFinish if we are rendering any layers using copybit.
+    // We call glFinish instead of locking the renderBuffer because the
+    // GPU could take longer than the genlock timeout value to complete
+    // rendering
+    glFinish();
+#endif
+
     for (size_t i=0; i<list->numHwLayers; i++) {
         if (list->hwLayers[i].compositionType == HWC_USE_COPYBIT) {
             retVal = drawLayerUsingCopybit(ctx, &(list->hwLayers[i]),
